src/core/ext/filters/client_channel/lb_policy/rls/rls.cc

```diff
@@ -1419,35 +1425,41 @@ void RlsLb::Cache::ResetAllBackoff() {
 void RlsLb::Cache::Shutdown() {
   map_.clear();
   lru_list_.clear();
-  grpc_timer_cancel(&cleanup_timer_);
+  if (cleanup_timer_handle_.has_value() &&
+      lb_policy_->channel_control_helper()->GetEventEngine()->Cancel(
+          *cleanup_timer_handle_)) {
+    if (GRPC_TRACE_FLAG_ENABLED(grpc_lb_rls_trace)) {
+      gpr_log(GPR_INFO, "[rlslb %p] cache cleanup timer canceled", lb_policy_);
+    }
+    cleanup_timer_handle_.reset();
```

Comment:
Maybe reset the handle regardless of whether `EventEngine::Cancel` returns true or false? I think technically we want to reset it when `EventEngine::Cancel` returns false.

Reply:
Done. I'm not sure that's necessary, but it doesn't hurt. Shutdown only appears to be called after RlsLb's `ShutdownLocked`, which means the callback will return early on `if (lb_policy->is_shutdown_) return;` regardless. The handle-reset in Cache::Shutdown is probably unnecessary, but safe, and conveys intent.

Reply:
oh, sorry then I missed that! Thanks for the explanation!